Mathlib/AlgebraicTopology/SimplicialSet/NonDegenerateSimplices.lean

```diff
@@ -0,0 +1,180 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.AlgebraicTopology.SimplicialSet.Degenerate
+import Mathlib.AlgebraicTopology.SimplicialSet.Simplices
+
+/-!
+# The partially ordered type of non degenerate simplices of a simplicial set
+
+In this file, we introduce the partially ordered type `X.N` of
+non degenerate simplices of a simplicial set `X`. We obtain
+an embedding `X.orderEmbeddingN : X.N ↪o X.Subcomplex` which sends
+a non degenerate simplex to the subcomplex of `X` it generates.
+
+Given an arbitrary simplex `x : X.S`, we show that there is a unique
+non degenerate `x.toN : X.N` such that `x.toN.subcomplex = x.subcomplex`.
+
+-/
+
+universe u
+
+open CategoryTheory Simplicial
+
+namespace SSet
+
+variable (X : SSet.{u})
+
+/-- The type of non degenerate simplices of a simplicial set. -/
+structure N extends X.S where mk'' ::
+  nonDegenerate : simplex ∈ X.nonDegenerate _
+
+namespace N
+
+variable {X}
+
+/-- Constructor in order to promote a simplex `s : X.S` into
+a term in `X.N`. -/
+@[simps toS]
+def mk' (s : X.S) (hs : s.simplex ∈ X.nonDegenerate _) : X.N where
+  toS := s
+  nonDegenerate := hs
```

Comment:
```suggestion
/-- The type of non degenerate simplices of a simplicial set. -/
structure N extends X.S where mk' ::
  nonDegenerate : simplex ∈ X.nonDegenerate _

namespace N

variable {X}
```

Is there any reason to have both `mk''` and `mk'`? They seem identical to me. You could state the dsimp lemma `mk'_toS` separately.